code/game/machinery/dish_drive.dm

```diff
@@ -0,0 +1,115 @@
+/obj/machinery/dish_drive
+	name = "dish drive"
+	desc = "A culinary marvel that uses matter-to-energy conversion to store dishes and shards. Convenient! \
+	Additional features include a vacuum function to suck in nearby dishes, and an automatic transfer beam that empties its contents into nearby disposal bins every now and then. \
+	Or you can just drop your plates on the floor, like civilized folk."
+	icon = 'goon/icons/obj/kitchen.dmi'
+	icon_state = "synthesizer"
+	idle_power_usage = 8 //5 with default parts
+	active_power_usage = 13 //10 with default parts
+	anchored = TRUE
+	density = FALSE
+	circuit = /obj/item/circuitboard/machine/dish_drive
+	var/static/list/item_types = list(/obj/item/trash/waffles,
+		/obj/item/trash/plate,
+		/obj/item/trash/tray,
+		/obj/item/reagent_containers/glass/bowl,
+		/obj/item/reagent_containers/food/drinks/drinkingglass,
+		/obj/item/kitchen/fork,
+		/obj/item/shard,
+		/obj/item/broken_bottle)
+	var/time_since_dishes = 0
+	var/suction_enabled = TRUE
+	var/transmit_enabled = TRUE
+
+/obj/machinery/dish_drive/Initialize()
+	. = ..()
+	RefreshParts()
+
+/obj/machinery/dish_drive/examine(mob/user)
+	..()
+	if(user.canUseTopic(src))
```

Comment:
Not quite sure why you'd need this check here, if it fails the user would be warned "You can't do that right now!", which seems a little strange for an examination message